onmt/inference_engine_ct2.py

```diff
@@ -0,0 +1,135 @@
+import ctranslate2
+import pyonmttok
+from onmt.constants import CorpusTask, DefaultTokens, ModelTask
+from onmt.inputters.dynamic_iterator import build_dynamic_dataset_iter
+from onmt.inputters.inputter import IterOnDevice
+from onmt.transforms import get_transforms_cls
+from onmt.utils.logging import logger
+
+
+class InferenceEngineCT2(object):
+
+    """Wrapper Class to run Inference with ctranslate2.
+
+    Args:
+        opt: inference options
+    """
+
+    def __init__(self, opt):
+        self.opt = opt
+        self.transforms_cls = get_transforms_cls(self.opt._all_transform)
+        self.logger = logger
+        if opt.world_size == 1:
+            self.device_id = 0
+
```

Comment:
do we really need these warm_up, build_ct2_translator, build_tokenizer ?
I think we can stick to the inference_engine.py logic and put eveything in __init__() but if there is a good reason then fine.